### PR TITLE
More scafacos related fixes

### PR DESCRIPTION
**Pull request by RudolfWeeber**
Fixes #

Description of changes:
 - 

PR Checklist
------------
 - [ ] Tests?
   - [ ] Interface
   - [ ] Core 
 - [ ] Docs?

### CODE REVIEW

File: src/python/espressomd/scafacos.pyx [hunk 1]
```diff
@@ -38,8 +45,45 @@ IF SCAFACOS == 1:
...
+                pname =p[i]
+                i+=1
+                
+                # The first item after the parameter name is alsways a value
+                # But in the case of array-like properties, there can also
+                # follow several values. Therefore, we treat the next
+                # words as part of the value, if they begin with a digit
+                pvalues=[p[i]]
+                i+=1
+                if i>=len(p):
```
Comment (KaiSzuttor) [inline src/python/espressomd/scafacos.pyx:62]: ~~this is already checked in the while condition~~

**Comment by RudolfWeeber** [comment]
On Thu, Nov 16, 2017 at 07:37:53AM +0000, Kai Szuttor wrote:
> KaiSzuttor commented on this pull request.
> 
> 
> 
> > -                res[p[2 * i + 1]] = p[2 * i + 2]
> +            
> +            
> +            method_params={}
> +            i=1
> +            while i<len(p):
> +                pname =p[i]
> +                i+=1
> +                
> +                # The first item after the parameter name is alsways a value
> +                # But in the case of array-like properties, there can also
> +                # follow several values. Therefore, we treat the next
> +                # words as part of the value, if they begin with a digit
> +                pvalues=[p[i]]
> +                i+=1
> +                if i>=len(p):
> 
> this is already checked in the while condition
But `i` is incremented since the head of the outer while loop, as
the first parameter value is consumed.

**Comment by KaiSzuttor** [comment]
You are right, i missed that

File: src/core/scafacos.cpp [hunk 2]
```diff
@@ -100,6 +100,8 @@ int ScafacosData::update_particle_data() {
 
 void ScafacosData::update_particle_forces() const {
   int it = 0;
+  if (positions.size()==0)
```
Comment (fweik) [inline src/core/scafacos.cpp]: `positions.empty()`

File: src/core/scafacos.cpp [hunk 3]
```diff
@@ -148,6 +150,7 @@ void ScafacosData::update_particle_forces() const {
   if (!dipolar()) {
     assert(it == fields.size());
   } else {
+    int tmp=positions.size()/3;
     assert(it = positions.size() / 3);
```
Comment (fweik) [inline src/core/scafacos.cpp]: not yours, but this should be `it == positions.size() / 3`.

**Review by fweik — APPROVED** [review]

**Pull request merged**